pkg/backend/filestate/backend.go

```diff
@@ -128,6 +111,53 @@ func New(d diag.Sink, u, stackConfigFile string) (Backend, error) {
 	}, nil
 }
 
+const filePathPrefix = "file://"
+
+// massageBlobPath takes the path the user provided and converts it to an appropriate form go-cloud
+// can support.  Importantly, s3/azblob/gs paths should not be be touched. This will only affect
+// file:// paths which have a few oddities around them that we want to ensure work properly.
+func massageBlobPath(path string) (string, error) {
+	if !strings.HasPrefix(path, filePathPrefix) {
+		// not a file:// path.  Keep this untouched and pass directly to gocloud.
+		return path, nil
+	}
+
+	// Strip off the "file://"" portion so we can examine and determine what to do with the rest.
+	path = strings.TrimPrefix(path, filePathPrefix)
+
+	// We need to specially handle ~.  The shell doesn't take care of this for us, and later
+	// functions we run into can't handle this either.
+	//
+	// From https://stackoverflow.com/questions/17609732/expand-tilde-to-home-directory
+	if strings.HasPrefix(path, "~") {
+		usr, err := user.Current()
+		if err != nil {
+			return "", errors.Wrap(err, "Could not determine current user to resolve `file://~` path.")
+		}
+
+		if path == "~" {
+			path = usr.HomeDir
+		} else {
+			path = filepath.Join(usr.HomeDir, path[2:])
+		}
+	}
+
+	// For file:// backend, ensure a relative path is resolved. fileblob only supports absolute paths.
+	path, err := filepath.Abs(path)
+	if err != nil {
+		return "", errors.Wrap(err, "An IO error occurred during the current operation")
```

Comment:
Nit - slightly strange error message for this case.

Reply:
It's an error we've had a while.  I don't think it's great.  But it doesn't bother me enough to do anything about it :)